app/src/main/java/org/p2p/wallet/infrastructure/network/data/RpcTransactionError.kt

```diff
@@ -0,0 +1,64 @@
+package org.p2p.wallet.infrastructure.network.data
+
+import com.google.gson.JsonArray
+import com.google.gson.JsonObject
+import org.p2p.wallet.utils.getOrNull
+
+/**
+ * For parsing structs like:
+ * ```json
+ * "err": {
+ *    "InstructionError": [
+ *        3,
+ *        { "Custom": 6022 }
+ *    ]
+ * }
+ *
+ *    or
+ * "err": {
+ *    "InstructionError": [
+ *        3,
+ *        "AccountInUse"
+ *    ]
+ * }
+ * ```
+ * @return valid typed RpcTransactionError or null if parsing is failed
+ */
+sealed interface RpcTransactionError {
+    companion object {
+        fun from(errJson: JsonObject): RpcTransactionError? {
+            // for example "InstructionError"
+            val transactionErrorType: String? = errJson.keySet().firstOrNull()
+            return when (transactionErrorType) {
```

Comment:
you can wrap it to some enum class

Reply:
no need for that, we know all the string values